docs/guide/custom-functions.md

```diff
@@ -46,25 +71,79 @@ import { FunctionPlugin } from 'hyperformula';
 // start creating a class
 class CountHF extends FunctionPlugin {
 
-// define functions inside this plugin
+  // define functions inside this plugin
   public static implementedFunctions = {
     'HYPER': {
-    // this method's functionality will be defined in the next step
+      // this method's functionality will be defined below
       method: 'hyper',
-    // optionally, mark your function as volatile
+      // set your optional parameters below
+      arrayFunction: false,
+      doesNotNeedArgumentsToBeComputed: false,
+      expandRanges: false,
+      isDependentOnSheetStructureChange: false,
       isVolatile: true,
+      repeatLastArgs: 4,
+      vectorizationForbidden: false,
+    }
+  };
+}
+```
+
+You can set the following optional parameters:
+
+| Option | Type | Description |
+| --- | --- | --- |
+| `arrayFunction` | Boolean | If set to `true`, the function enables the [array arithmetic mode](arrays.md) in its arguments and nested expressions. |
+| `doesNotNeedArgumentsToBeComputed` | Boolean | If set to `true`, the function treats reference or range arguments as arguments that don't create dependency.<br><br>Other arguments are properly evaluated. |
+| `expandRanges` | Boolean | If set to `true`, ranges in the function's arguments are inlined to (possibly multiple) scalar arguments. |
+| `isDependentOnSheetStructureChange` | Boolean | If set to `true`, the function gets recalculated with each sheet shape change. |
```

Comment:
Maybe we should mention defaults? Does all our functions are recalculated on sheet shape change? What does the "shape change" mean? I guess, adding/removing rows or cols

Reply:
@wojciechczerniak 

Shape change: done edf99e9

Defaults: @izulin what are the above options' default values? (I'm guessing `false` for all booleans)